stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt

```diff
@@ -97,14 +97,36 @@ public interface ChatDomain {
     public fun getVersion(): String
 
     public data class Builder(
-        private var appContext: Context,
-        private var client: ChatClient,
-        private var user: User? = null
+        private val appContext: Context,
+        private val client: ChatClient,
     ) {
+        private var user: User? = null
 
-        public constructor(client: ChatClient, appContext: Context) : this(appContext, client, null)
+        public constructor(client: ChatClient, appContext: Context) : this(appContext, client)
 
-        public constructor(client: ChatClient, user: User?, appContext: Context) : this(appContext, client, user)
+        @Deprecated(
+            message = "Use constructor without user",
+            replaceWith = ReplaceWith("Use ChatDomain.Builder(appContext, chatClient) instead")
+        )
+        public constructor(appContext: Context, client: ChatClient, user: User?) : this(appContext, client) {
+            this.user = user
+        }
+
+        @Deprecated(
+            message = "Use constructor without user",
+            replaceWith = ReplaceWith("Use ChatDomain.Builder(appContext, chatClient) instead")
+        )
+        public constructor(client: ChatClient, appContext: Context, user: User?) : this(appContext, client) {
+            this.user = user
+        }
```

Comment:
This looks like a brand new constructor that we didn't have before

Reply:
Previously it was the main one

Reply:
That was context/client/user, which we have above. This one is client/context/user.

It's new according to the API diff as well:

```diff
+ public fun <init> (Lio/getstream/chat/android/client/ChatClient;Landroid/content/Context;Lio/getstream/chat/android/client/models/User;)V
```

Reply:
Oh, I see